logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/GetUserE2EICertificateUseCase.kt

```diff
@@ -31,12 +33,13 @@ interface IsOtherUserE2EIVerifiedUseCase {
 
 class IsOtherUserE2EIVerifiedUseCaseImpl internal constructor(
     private val mlsConversationRepository: MLSConversationRepository,
-    private val isE2EIEnabledUseCase: IsE2EIEnabledUseCase
+    private val isE2EIEnabledUseCase: IsE2EIEnabledUseCase,
+    private val userRepository: UserRepository
 ) : IsOtherUserE2EIVerifiedUseCase {
     override suspend operator fun invoke(userId: UserId): Boolean =
         if (isE2EIEnabledUseCase()) {
-            mlsConversationRepository.getUserIdentity(userId).fold({ false }, { it.isUserMLSVerified() }
-            )
+            val nameHandle = userRepository.getNameAndHandler(userId).getOrNull()
+            mlsConversationRepository.getUserIdentity(userId).fold({ false }, { it.isUserMLSVerified(nameHandle) })
         } else {
             false
         }
```

Comment:
suggestion:
`
override suspend operator fun invoke(userId: UserId): Boolean =
    isE2EIEnabledUseCase() && 
    userRepository.getNameAndHandler(userId).getOrNull()?.let { nameHandle ->
        mlsConversationRepository.getUserIdentity(userId).fold(
            onSuccess = { it.isUserMLSVerified(nameHandle) },
            onFailure = { false }
        )
    } ?: false
`

Reply:
I tried even shorter:
```
override suspend operator fun invoke(userId: UserId): Boolean =
    isE2EIEnabledUseCase() && userRepository.getNameAndHandler(userId).flatMap { nameHandle ->
        mlsConversationRepository.getUserIdentity(userId).map { it.isUserMLSVerified(nameHandle) }
    }.getOrElse(false)
```
but i don't like it. Yes it's 3 lines instead of 6, but IMO `if else` is more readable and easier to understand in that case :)